src/Integrators.jl

```diff
@@ -222,8 +239,25 @@ end
 # Dipole integration
 ################################################################################
 
-@inline rhs_dipole(s, B) = -s × B
-@inline rhs_dipole(s, B, λ) = -s × (B + λ * (s × B))
+
+@inline function rhs_dipole!(Δs, s, ξ, ∇E, integrator)
+    (; Δt, λ, kT) = integrator
+    if iszero(λ) && iszero(kT)
+        @. Δs = -s × (- Δt*∇E)
+    else
+        @. Δs = -s × (- Δt*∇E + ξ - Δt*λ*(s × ∇E))
+    end
+end
+
+function rhs_sun!(ΔZ, Z, ξ, HZ, integrator)
+    (; kT, λ, Δt) = integrator
+
+    @. ΔZ = - Δt*(im+λ)*HZ + ξ
+    if !iszero(λ) || !iszero(kT)
```

Comment:
Is the part after the `||` necessary? We should never have the case that `λ=0` but `kT>0`.

Reply:
I could imagine having that case! It represents the weak-coupling limit :) so for example if I'm running ImplicitMidpoint at finite temperature, but then want to compare to the uncoupled case (which is what Sunny does by default), I would probably just set lambda = 0 (still with finite kT) to achieve this

Reply:
I see. Just to be clear, I'm not here saying we disallow users from setting a finite `kT` with `λ=0`. My point is just that the fast path should be taken whenever `λ=0` (since the kT will not practically contribute in the impelmentation). This will be the case if the OR is removed.